XTrophy.cs

```diff
@@ -92,6 +96,77 @@ public Int64 MyBalance()
             } else Error.Throw("XTrophyError: Operation denied");
         }
 
+
+        //// Withdrawal methods
+
+        // Freeze XTrophy.
+        public void WithdrawalRequest(Int64 amount, String hashCardNumber)
+        {
+            Require(amount > 0, "Amount must be positive");
+
+            Bytes sender = Info.Sender();
+            WithdrawalData withdrawalData = FrozenBalance.GetOrDefault(sender, new WithdrawalData());
+            Require(withdrawalData.amount == 0, "Withdrawal is already there");
+
+            Int64 balance = Balance.GetOrDefault(sender, 0);
+            Require(balance >= amount, "Not enough XTrophy for Withdrawal");
+
+            Balance[sender] = balance - amount;
+            FrozenBalance[sender] = new WithdrawalData(amount, hashCardNumber);
+            Log.Event("WithdrawalRequest", new EventData(sender, amount));
+        }
+
+        // Burn frozen XTrophy.
+        public void WithdrawalComplete(Bytes address)
+        {
+            assertIsOwner();
+
+            nonEmptyFrozenBalance(address);
+
+            Int64 frozenBalance = FrozenBalance[address].amount;
+            FrozenBalance[address] = new WithdrawalData();
```

Comment:
May be set to null? @Chicker Looks like we forgot to add Remove method to Dictionary.